src/UBAFeeCalculator/UBAFeeSpokeCalculator.ts

```diff
@@ -60,17 +62,30 @@ export default class UBAFeeSpokeCalculator {
       return this.cachedRunningBalance[key];
     }
 
+    // Attempt to resolve the trigger hurdle to include in the running
+    // balance calculation
+    const triggerHurdle = this.getBalanceTriggerThreshold();
+
     // If the last validated running balance is undefined, we need to compute the running balance from scratch
     // This is the case when the UBA Fee Calculator is first initialized or run on a range
     // that we haven't computed the running balance for yet
     const historicalResult = this.recentRequestFlow.slice(startIdx, endIdx).reduce((acc, flow) => {
+      let resultantValue = acc;
       if (isUbaInflow(flow)) {
-        return acc.add(toBN(flow.amount));
+        resultantValue = acc.add(toBN(flow.amount));
       } else if (isUbaOutflow(flow)) {
-        return acc.sub(toBN(flow.amount));
-      } else {
-        return acc;
+        resultantValue = acc.sub(toBN(flow.amount));
+      }
+
+      // If the trigger hurdle is surpassed, we need to return the trigger hurdle value
```

Comment:
What about a bottom threshold whereby we go below that running balance and need the hub pool to transfer funds into the spoke pool? Do we handle that case?

Reply:
I don't believe we have this case anymore as of today. I was told by @nicholaspai that our primary thresholds are when the running balance exceeds a target.

Reply:
I'm not sure I agree. Mechanically, imagine all spoke pools start with 0 balance. I make a single transfer from Optimism to Arbitrum for 100 USDC. Let's assume, for simplicity that all fees are 0 (including balancing fee -- we can choose to have the curve be flat).

My deposit makes the running balance on Optimism 100 USDC. The relayer relays my transaction on arbitrum, sending me 100 USDC. They now need to be refunded 100 USDC on arbitrum in the next bundle, but the arbitrum spoke pool needs to be sent 100 USDC from the HubPool to fulfil that liability. How would that happen if no lower threshold exists?

Reply:
Fair point! Resolved here: https://github.com/across-protocol/sdk-v2/pull/196/commits/1b468d46e6f0ed4a2f0d91e6cdcb7fb3e6c70f4c